test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -2050,6 +2054,177 @@ func TestBindingRemoval(t *testing.T) {
+		{
+			name:       "principal is not authorized for custom verb on current resource",
+			expression: "authorizer.requestResource.check('anthropomorphize').allowed()",
+			allowed:    false,
+		},
+		{
+			name:           "serviceaccount is authorized for custom verb on current resource",
+			extraAccountFn: serviceAccountClient("default", "extra-acct"),
+			extraAccountRbac: &rbacv1.PolicyRule{
+				Verbs:     []string{"anthropomorphize"},
+				APIGroups: []string{""},
+				Resources: []string{"pods"},
+			},
+			expression: "authorizer.serviceAccount('default', 'extra-acct').group('').resource('pods').check('anthropomorphize').allowed()",
+			allowed:    true,
+		},
+	}
+
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			clients := map[string]func(t *testing.T, adminClient *clientset.Clientset, clientConfig *rest.Config, rules []rbacv1.PolicyRule) *clientset.Clientset{
+				"user":           secondaryAuthorizationUserClient,
+				"serviceaccount": secondaryAuthorizationServiceAccountClient,
+			}
+
+			for clientName, clientFn := range clients {
+				t.Run(clientName, func(t *testing.T) {
+					defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ValidatingAdmissionPolicy, true)()
+					server, err := apiservertesting.StartTestServer(t, nil, []string{
+						"--enable-admission-plugins", "ValidatingAdmissionPolicy",
+						"--authorization-mode=RBAC",
+						"--anonymous-auth",
+					}, framework.SharedEtcd())
+					if err != nil {
+						t.Fatal(err)
+					}
+					defer server.TearDownFn()
+
+					// For test set up such as creating policies, bindings and RBAC rules.
+					adminClient := clientset.NewForConfigOrDie(server.ClientConfig)
+
+					// Principal is always allowed to create and update namespaces so that the admission requests to test
+					// authorization expressions can be sent by the principal.
+					rules := []rbacv1.PolicyRule{{
+						Verbs:     []string{"create", "update"},
+						APIGroups: []string{""},
+						Resources: []string{"namespaces"},
+					}}
+					if testcase.rbac != nil {
+						rules = append(rules, *testcase.rbac)
+					}
+
+					client := clientFn(t, adminClient, server.ClientConfig, rules)
+
+					if testcase.extraAccountFn != nil {
+						var extraRules []rbacv1.PolicyRule
+						if testcase.extraAccountRbac != nil {
+							extraRules = append(rules, *testcase.extraAccountRbac)
+						}
+						testcase.extraAccountFn(t, adminClient, server.ClientConfig, extraRules)
+					}
+
+					policy := withWaitReadyConstraintAndExpression(withValidations([]admissionregistrationv1alpha1.Validation{
+						{
+							Expression: testcase.expression,
+						},
+					}, withFailurePolicy(admissionregistrationv1alpha1.Fail, withNamespaceMatch(makePolicy("validate-authz")))))
+					if _, err := adminClient.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), policy, metav1.CreateOptions{}); err != nil {
```

Comment:
Why the TODO?

Reply:
The `context.TODO()` ? It is idiomatic to use this when supplying a context in tests like this.